packages/solidjs/src/errorboundary.ts

```diff
@@ -0,0 +1,31 @@
+import { captureException } from '@sentry/browser';
+import type { Component, JSX } from 'solid-js';
+import { mergeProps, splitProps, untrack } from 'solid-js';
+import { createComponent } from 'solid-js/web';
+
+type ErrorBoundaryProps = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  fallback: JSX.Element | ((err: any, reset: () => void) => JSX.Element);
+  children: JSX.Element;
+};
+
+/**
+ * A higher-order component to wrap Solid's ErrorBoundary to capture exceptions.
+ */
+export function withSentryErrorBoundary(ErrorBoundary: Component<ErrorBoundaryProps>): Component<ErrorBoundaryProps> {
+  const SentryErrorBoundary = (props: ErrorBoundaryProps): JSX.Element => {
+    const [local, others] = splitProps(props, ['fallback']);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const fallback = (error: any, reset: () => void): JSX.Element => {
+      captureException(error);
+
+      const f = local.fallback;
+      return typeof f === 'function' ? untrack(() => f(error, reset)) : f;
```

Comment:
can we add a comment here as to why we need to call `untrack`?

Reply:
Actually, it's probably not needed. I initially checked how solid implements error boundaries and they use `untrack` on the fallback, but they use it within a `createMemo`. I _believe_ we don't need to do that since we are not yet in a reactive context here.

I removed it and reran my tests and local example app and cannot tell a difference.